packages/vm/src/cache.rs

```diff
@@ -182,6 +190,19 @@ where
         self.inner.lock().unwrap().stats
     }
 
+    pub fn heavy_metrics(&self) -> HeavyMetrics {
```

Comment:
"heavy metrics" sounds like a superset of metrics for me. I think we can just call it "pinned metrics" as a shorthand for "pinned contract cache metrics". Many users don't use pinned contracts at all. So its probably good to keep the call specific to that feature.